aten/src/ATen/native/quantized/cpu/Pooling.cpp

```diff
@@ -94,6 +95,80 @@ void spatial_dilated_max_pooling(
   });
 }
 
+template <typename T>
+void spatial_dilated_max_pooling3d(
+    const T* iData,
+    int64_t iC, // input/output channels
+    int64_t iT,
+    int64_t iH,
+    int64_t iW, // input sizes
+    int64_t oT,
+    int64_t oH,
+    int64_t oW, // output sizes
+    int64_t kT,
+    int64_t kH,
+    int64_t kW, // kernel size
+    int64_t sT,
+    int64_t sH,
+    int64_t sW, // strides
+    int64_t pT,
+    int64_t pH,
+    int64_t pW, // padding
+    int64_t dT,
+    int64_t dH,
+    int64_t dW, // dilation
+    T* oData) { // output arrays (data and max-index)
+  // Handle each bs
+  at::parallel_for(0, iC, 0, [&](int64_t start, int64_t end) {
```

Comment:
One more thing, if we care about channels first performance. Even parallel on N*C is not enough, say N=1, C=1 and #cores = 4.

if we do not care about CF perf, that's fine.
if we do care, then parallel on NCTH and cache the output indices from W, going to be much faster.

Reply:
Thanks for the suggestions. Add a TODO in `spatial_dilated_max_pooling3d` as what we can do to further the CF performance.